radarlabs/radar-sdk-ios
# Make geofence geometry optional

Interface change:

We modify the interface of `searchGeofences` to the following

```
/**
Gets the device's current location, then searches for geofences near that location, sorted by distance.
 @param completionHandler A completion handler.

 @see https://radar.com/documentation/api#search-geofences
 */
+ (void)searchGeofences:(RadarSearchGeofencesCompletionHandler)completionHandler NS_SWIFT_NAME(searchGeofences(tags:metadata:completionHandler:));

/**
 Searches for geofences near a location, sorted by distance.

 @param near The location to search. Use null to search near the device's current location.
 @param radius The radius to search, in meters. A number between 100 and 10000. If -1 is entered, the server defaults to using unlimited radius.
 @param tags An array of tags to filter. See https://radar.com/documentation/geofences
 @param metadata A dictionary of metadata to filter. See https://radar.com/documentation/geofences
 @param limit The max number of geofences to return. A number between 1 and 1000. Recommend to be set to 100.
 @param includeGeometry A boolean indicating whether to include geometry in the response. Recommend to be set to false.
 @param completionHandler A completion handler.

 @see https://radar.com/documentation/api#search-geofences
 */
 + (void)searchGeofencesNear:(CLLocation *_Nullable)near
                     radius:(int)radius
                       tags:(NSArray<NSString *> *_Nullable)tags
                   metadata:(NSDictionary *_Nullable)metadata
                      limit:(int)limit
            includeGeometry:(BOOL)includeGeometry
          completionHandler:(RadarSearchGeofencesCompletionHandler)completionHandler NS_SWIFT_NAME(searchGeofences(near:radius:tags:metadata:limit:includeGeometry:completionHandler:));


```
We essentially have 2 methods, a bare bones one that should be easy to use, and a maximal one with optional named arguments for more advanced use-cases. 


QA:
We were able to observe successful calls with the example app code snippet to both `getContext` and `searchGeofence`
```
let location = CLLocation(latitude: 49.68956683045484, longitude: -107.27634218307078)
Radar.getContext(location:location) { (status, location, context) in
  print("Context: status = \(Radar.stringForStatus(status)); location = \(String(describing: location)); context?.geofences = \(String(describing: context?.geofences)); context?.place = \(String(describing: context?.place)); context?.country = \(String(describing: context?.country))")
        }
Radar.searchGeofences() { (status, location, geofences) in
            print("Search geofences: status = \(Radar.stringForStatus(status)); geofences = \(String(describing: geofences))")
        }
        
Radar.searchGeofences(near: location, radius: -1, tags: nil, metadata: nil, limit: 100, includeGeometry: true) { (status, location, geofences) in
            print("Search geofences 2: status = \(Radar.stringForStatus(status)); geofences = \(String(describing: geofences))")
        }
```

with the printed output
<img width="517" alt="image" src="https://github.com/radarlabs/radar-sdk-ios/assets/139801512/a0077452-1635-4cbe-94ac-5182a0e0f0a2">
